go/fn/subobject.go

```diff
@@ -35,3 +36,55 @@ func (o *SubObject) GetMap(k string) *SubObject {
 	}
 	return &SubObject{obj: m}
 }
+
+// GetSlice gets the component in the `fields` path. It expects the component to be
+// a slice of SubObjects.
+func (o *SubObject) GetSlice(fields ...string) (SliceSubObjects, bool, error) {
```

Comment:
Nit: How about if we name this `GetNestedSlice` for consistency with unstructured, and save `GetSlice` for the one-key-at-a-time version?

Reply:
I'm a little bit worried this may confuse users because
1. GetNestedXXX is a internal.MapVariant naming convention
2. KubeObject naming convention (that calls GetSlice) is GetXXX and GetXXXOrDie (e.g. GetInt GetString). all accepts `fields` as a slice of strings. 
3. Both KubeObject (KRM resource) and SubObject (sub component of KRM resources) can call `GetSlice`. 

qq: what would be the problem if users use the same function for a one-key-at-a-time and concatenated field paths usage? I"m thinking of filtering the return element by key value pairs in the form of `GetSlice("key=value")`.

Reply:
I'm leaning towards making the names consistent with `unstructured` package. But I guess we can use it no matter if the fields are one or multiple.

IIRC when I did it initially, names with pattern `GetNestedXXX` were already taken in kyaml. So I use different names.

Reply:
If we want to do the renaming, we should do it everywhere in the SDK. I guess we can do that in a followup PR.

Reply:
That's a good points to borrow the `unstructured` naming convention. I'm going to change the functions which takes a slice of strings as "fields" to from `GetXXX` to `NestedXXX` and `SetXXX` to `SetNestedXXX` and renaming the `GetSliceN` to `GetNestedSlice`.